### DISCUSSION TITLE
How to provide GitHub credentials to containers?

### DISCUSSION THREAD

**Discussion by thevedicdeveloper** [devcontainers/community#58] [Q&A]
I want to clone a repository from inside the container.

the container is just a basic Alpine image
```dockerfile
FROM alpine:latest
```
Since it's new, I have to install Git first. After doing that, when I try to clone my private repository.

It asks for first **user.name** and **user.email**. I provided that as well.

But when I am trying to push, it is saying I am not authenticated.

How can I provide my git credentials to my container? There are a couple of options like **gpg key**, and **access token**.

**Comment by bamurtaugh** [comment]
Thanks for sharing this @thevedicdeveloper! Depending on your setup, the tips here may help: https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials.

**Comment by thevedicdeveloper** [comment]
Thanks for sharing @bamurtaugh
